api/src/main/java/org/apache/gravitino/rel/Table.java

```diff
@@ -44,6 +44,28 @@
 @Evolving
 public interface Table extends Auditable {
 
+  /**
+   * The reserved property name for the table format. This property indicates the format of the
+   * table, such as "iceberg", "hudi", "lance", etc. Some catalog implementations may use this
+   * property to determine the table format and perform specific operations based on the format.
+   */
+  String PROPERTY_TABLE_FORMAT = "format";
+
+  /**
+   * The reserved property name to indicate whether the table is external. This property is a
+   * boolean value represented as a string ("true" or "false"). if true (the table is external), the
+   * drop operation will not delete the underlying data.
+   */
+  String PROPERTY_EXTERNAL = "external";
+
+  /**
+   * The reserved property name for the table location. This property indicates the physical
+   * location of the table's data, such as a file path or a URI. The location can be specified when
+   * creating the table. If not, the catalog implementation may use a location based on the catalog
+   * and schema location properties.
+   */
+  String PROPERTY_LOCATION = "location";
```

Comment:
Do we need to mark `location` as an optional reserved property in `catalog` and `schema`?